pkg/ebpf/mappings.go

```diff
@@ -296,7 +296,10 @@ func AddProbeFDMappings(mgr *manager.Manager) {
 	mappingLock.Lock()
 	defer mappingLock.Unlock()
 
-	for _, p := range mgr.Probes {
+	// GetProbes returns a copy of the probes. We use this because mgr.Probes is mutable
+	// and so may change from underneath us.
+	probes := mgr.GetProbes()
```

Comment:
did we consider using an rwlock on the probes in all places we need access and copy them to avoid the race?

Reply:
it is more costly, in terms of memory, but perhaps this is not significant given the average size of the array and the frequency of those methods being called.

since you mentioned that you considered the other approach and decided to choose the copy solution, i'm closing this comment